core/services/pipeline/common.go

```diff
@@ -289,6 +289,16 @@ func (trrs *TaskRunResults) GetNextTaskOf(task TaskRunResult) *TaskRunResult {
 	return nil
 }
 
+func (trrs TaskRunResults) AllErrors() error {
+	var errs []error
+	for _, trr := range trrs {
+		if trr.Result.Error != nil {
+			errs = append(errs, trr.Result.Error)
+		}
+	}
+	return errors.Join(errs...)
```

Comment:
nit/ could simplify this a bit:
```suggestion
func (trrs TaskRunResults) AllErrors() (err error) {
	for _, trr := range trrs {
		if trr.Result.Error != nil {
			err = errors.Join(err, trr.Result.Error)
		}
	}
	return
```